src/wasm/wasm-binary.cpp

```diff
@@ -557,40 +554,50 @@ void WasmBinaryWriter::writeTableElements() {
     return;
   }
 
-  BYN_TRACE("== writeTableElements\n");
+  std::set<Name> referencedFuncs;
+
+  BYN_TRACE("== writeElementSegments\n");
   auto start = startSection(BinaryConsts::Section::Element);
   o << U32LEB(elemCount);
 
-  for (auto& table : wasm->tables) {
-    for (auto& segment : table->segments) {
-      Index tableIdx = getTableIndex(table->name);
-      // No support for passive element segments yet as they don't belong to a
-      // table.
-      bool isPassive = false;
-      bool isDeclarative = false;
-      bool hasTableIndex = tableIdx > 0;
-      bool usesExpressions = false;
-
-      uint32_t flags =
-        (isPassive ? BinaryConsts::IsPassive |
-                       (isDeclarative ? BinaryConsts::IsDeclarative : 0)
-                   : (hasTableIndex ? BinaryConsts::HasIndex : 0)) |
-        (usesExpressions ? BinaryConsts::UsesExpressions : 0);
-
-      o << U32LEB(flags);
+  for (auto& segment : wasm->elementSegments) {
+    Index tableIdx = 0;
+
+    bool isPassive = segment->table.isNull();
+    bool usesExpressions = false;
```

Comment:
It looks like this is never changed from `false`?

Reply:
Correct. I just wanted to this part to follow the complete spec, and then finish it in the next PR when I'll be changing element segments to have expressions instead of just function names. I can remove this for now if you want.

Reply:
Would be good to either remove or add a TODO comment explaining the future intentions.

Reply:
Done